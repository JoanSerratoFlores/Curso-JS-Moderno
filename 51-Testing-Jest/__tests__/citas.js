import Citas from "../js/classes/Citas";

describe('Probar la clase de citas', () => {
    
    const citas = new Citas();
    
    const id = Date.now();
    
    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Juan',
            telefono: '812934567',
            fecha: '10-12-2020',
            hora:'10:30',
            sintomas: 'Solo Duerme'
        };

        citas.agregarCita(citaObj);

        //Prueba
        expect(citas).toMatchSnapshot();

    })

    test('actualizar cita', () => {
        const citaObjActual = {
            id,
            mascota: 'Nuevo Nombre',
            propietario: 'Juan',
            telefono: '812934567',
            fecha: '10-12-2020',
            hora:'10:30',
            sintomas: 'Solo Duerme'
        };

        citas.editarCita(citaObjActual)

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    });

})