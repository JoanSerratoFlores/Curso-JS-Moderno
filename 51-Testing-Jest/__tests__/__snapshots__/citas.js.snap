// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1682670990921,
      "mascota": "Hook",
      "propietario": "Juan",
      "sintomas": "Solo Duerme",
      "telefono": "812934567",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`Probar la clase de citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1682670990921,
      "mascota": "Nuevo Nombre",
      "propietario": "Juan",
      "sintomas": "Solo Duerme",
      "telefono": "812934567",
    },
  ],
}
`;
